### PR TITLE
python3

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

**Comment by asottile** [comment]
@bukzor should be good to go now, all the tests pass

File: testify/test_case.py [hunk 1]
```diff
@@ -66,12 +68,8 @@ def _cmp_str(instance):
         """Return a canonical representation of a TestCase for sorting and hashing."""
         return "%s.%s" % (instance.__module__, instance.__name__)
 
-    def __cmp__(cls, other):
```
Comment (eevee) [inline testify/test_case.py]: was this important?

**Comment by eevee** [comment]
looks ok at a glance

File: testify/test_case.py (hunk 1 above)
Comment (asottile) [inline testify/test_case.py]: nah, a remnant of past times I believe.  It was getting used in py2 for ==, but reference equality for classes is usually good enough :) in py3 it was causing flake8 to balk at `cmp`

Comment (eevee) [inline testify/test_case.py]: i ask because the docstring mentions _sorting_, and preserving sort order seems like a useful thing to be doing  :)

Comment (asottile) [inline testify/test_case.py]: I was actually the one who made this obsolete with this change here: https://github.com/Yelp/Testify/pull/216/files#diff-e4d2bf0d20a246b72b742afc1d4c3560R102

File: testify/test_case.py [hunk 2]
```diff
@@ -15,25 +15,27 @@
 
 """This module contains the TestCase class and other helper code, like decorators for test methods."""
 
-# TODO: finish doing the retry stuff for the inner clauses
+from __future__ import absolute_import
 
+# TODO: finish doing the retry stuff for the inner clauses
```
Comment (bukzor) [inline testify/test_case.py]: is this a shipit blocker?

File: testify/test_case.py [hunk 3]
```diff
@@ -142,8 +139,14 @@ def __init__(self, *args, **kwargs):
         # for now, we still support the use of unittest-style assertions defined on the TestCase instance
         for name in dir(deprecated_assertions):
             if name.startswith(('assert', 'fail')):
-                setattr(self, name, instancemethod(getattr(deprecated_assertions, name), self, self.__class__))
-
+                setattr(
+                    self,
+                    name,
+                    # http://stackoverflow.com/q/4364565
+                    getattr(deprecated_assertions, name).__get__(
```
Comment (bukzor) [inline testify/test_case.py]: could name this as compat.instancemethod, to be a bit more clear.

p4

**Comment by bukzor** [comment]
:lgtm:

I'd like a shipit from one of #releng, given the size of this change.
@milki @bis12

**Comment by ymilki** [comment]
This looks great! Thanks for going through all the code. python3 here we come!

**Pull request merged**